lib/topology_lib_sflowtool/parser.py

```diff
@@ -78,49 +77,169 @@ def parse_sflowtool(raw_output):
+                              'src_ip', 'dst_ip', 'ip_protocol', 'ip_tos',
+                              'ip_ttl', 'icmp_type', 'icmp_code', 'tcp_flags',
+                              'packet_size', 'ip_size', 'sampling_rate']
+        sample_packet_fields = ['packet_type', 'agent_address', 'if_index',
+                                'if_type', 'if_speed', 'if_direction',
+                                'if_status', 'in_octets', 'in_ucastPkts',
+                                'in_mcastPkts', 'in_bcastPkts', 'in_discards',
+                                'in_errors', 'in_unknownProtos', 'out_octets',
+                                'out_ucastPkts', 'out_mcastPkts',
+                                'out_bcastPkts', 'out_discards', 'out_errors',
+                                'if_promiscuousMode']
+
+        output = raw_output.splitlines()
+        flow_count = 0
+        sample_count = 0
+        result = {}
+        packets = []
+
+        for line in output:
+            packet = {}  # sFlow packet information
+            sflow_packet = line.split(",")
+            if sflow_packet[0] == 'FLOW':
+                assert len(sflow_packet) == len(flow_packet_fields)
+                for field in range(len(sflow_packet)):
+                    packet[flow_packet_fields[field]] = sflow_packet[field]
+                flow_count = flow_count + 1
+                packets.append(packet)
+            elif sflow_packet[0] == 'CNTR':
+                assert len(sflow_packet) == len(sample_packet_fields)
+                for field in range(len(sflow_packet)):
+                    packet[sample_packet_fields[field]] = sflow_packet[field]
+                sample_count = sample_count + 1
+                packets.append(packet)
+
+        result['flow_count'] = flow_count
+        result['sample_count'] = sample_count
+        result['packets'] = packets
+        return result
+
+    elif mode == 'detail':
+
+        start_datagram = 'startDatagram =================================\n'
+        end_datagram = 'endDatagram   =================================\n'
+
+        start_sample = 'startSample ----------------------\n'
+        end_sample = 'endSample   ----------------------\n'
+
+        finder = r'{}(.*?){}'
+
+        datagrams = findall(
+            finder.format(start_datagram, end_datagram), raw_output, DOTALL)
+
+        result = []
+        attribute_re = '((?!startSample|endSample)(.+?) (.+))'
+        start_sample_re = 'startSample .+'
+        end_sample_re = 'endSample .+'
+
+        for datagram in datagrams:
+
+            result.append(OrderedDict())
+            datagram_lines = datagram.splitlines()
+            sample_flag = False
+
+            for datagram_line in datagram_lines:
+
+                attribute = match(attribute_re, datagram_line)
+                start_sample = match(start_sample_re, datagram_line)
+                end_sample = match(end_sample_re, datagram_line)
```

Comment:
Can you try using the findall() for identifying samples as well? Just like how it was used for finding datagrams. It is a little confusing to understand what is going on in this for loop.